toolchain/check/testdata/builtin_conversions/no_prelude/convert_class_type_to_generic_facet_value.carbon

```diff
@@ -0,0 +1,403 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/builtin_conversions/no_prelude/convert_class_type_to_generic_facet_value.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/builtin_conversions/no_prelude/convert_class_type_to_generic_facet_value.carbon
+
+interface Generic(Scalar:! type) {
+  fn F();
+}
+
+class GenericParam {}
+
+class ImplsGeneric {}
+impl ImplsGeneric as Generic(GenericParam) {
+  fn F() {}
+}
+
+interface Other {
+  fn G();
+}
+impl ImplsGeneric as Other {
+  fn G();
+}
+
+fn CallGenericMethod(T:! type, U:! Generic(T)) {}
+
+fn G() {
+  CallGenericMethod(GenericParam, ImplsGeneric);
+}
+
+fn PassThroughToGenericMethod(T:! type, U:! Generic(T)) {
+  CallGenericMethod(T, U);
+}
+
+fn H() {
+  PassThroughToGenericMethod(GenericParam, ImplsGeneric);
+}
```

Comment:
I think it'd be good to also test the case where `T` is deduced after `U`, so we can't do the conversion eagerly:
```carbon
fn CallGenericMethod[T:! type](U:! Generic(T), t: T) {}

fn G(t: GenericParam) {
  CallGenericMethod(GenericParam, t);
}
```